pkg/virt-launcher/virtwrap/api/schema.go

```diff
@@ -661,29 +662,50 @@ type InterfaceTarget struct {
 }
 
 type Alias struct {
-	Name string `xml:"name,attr"`
+	name        string
+	userDefined bool
 }
 
-type UserAlias Alias
+// Package private, responsible to interact with xml.Encoder and xml.Decoder
+type userAliasXML struct {
+	Name        string `xml:"name,attr"`
+	UserDefined bool   `xml:"userdefined,attr,omitempty"`
+}
 
 type Rom struct {
 	Enabled string `xml:"enabled,attr"`
 }
 
+func NewAlias(aliasName string) *Alias {
+	return &Alias{name: aliasName}
+}
+
+func NewUserDefinedAlias(aliasName string) *Alias {
+	return &Alias{name: aliasName, userDefined: true}
+}
+
+func (alias Alias) GetName() string {
+	return alias.name
+}
+
 func (alias Alias) MarshalXML(e *xml.Encoder, start xml.StartElement) error {
-	userAlias := UserAlias(alias)
-	userAlias.Name = UserAliasPrefix + userAlias.Name
-	return e.EncodeElement(userAlias, start)
+	userAliasXml := userAliasXML{Name: alias.name, UserDefined: alias.userDefined}
```

Comment:
The public visible part looks fine but you have to change `MarshalXML` and `UnmarshalXML`: We have to provide the same content like before the libvirt. We can't just invent a new boolean field in the xml.

Consider something like this for marshalling:

```go
func (alias Alias) MarshalXML(e *xml.Encoder, start xml.StartElement) error {
        name := alias.name
        if alias.userDefined {
          name = UserAliasPrefix + name
        }
	start.Attr = []xml.Attr{xml.Attr{Name: xml.Name{Local: "name"}, Value: name}}
	e.EncodeToken(start)
	e.EncodeToken(xml.EndElement{Name: start.Name})
}
```

And vice-versa:

```go
func (alias *Alias) UnmarshalXML(d *xml.Decoder, start xml.StartElement) error {
	var name string
	for _ ,attr := range start.Attr {
		if attr.Name.Local == "name" {
			name = attr.Value
			break
		}
	}
	if strings.HasPrefix(name, UserAliasPrefix) {
		alias.UserDefined = true
	}
	alias.Name = strings.TrimPrefix(name)
	return nil
}
```